File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

float d;
int c;
int coins;


int main(void)
{
// prompt user to insert cash  
    do
    {
        d = get_float("Change owed: ");
    }
    while (d <= 0.001);

// translating dollars to cents
    c = round(d * 100);
// introducing variable coins
    coins = 0;

// quarters
    while (c - 25 >= 0)
    {
        c = c - 25;
        coins++;
    }
// dimes
    while (c - 10 >= 0)
    {
        c = c - 10;
        coins++;
    }
// nickels
    while (c - 5 >= 0)
    {
        c = c - 5;
        coins++;
    }
// cents
    while (c - 1 >= 0)
    {
        c = c - 1;
        coins++;
    }
// printout of results
    printf("%i\n", coins);
}

File: cash.py
from cs50 import get_float

while True:
    cash = get_float("chang owed: ")
    if cash > 0:
        break

cash = round(cash * 100)
coins = 0

while (cash - 25 >= 0):
    cash -= 25
    coins += 1

while (cash - 10 >= 0):
    cash -= 10
    coins += 1

while (cash - 5 >= 0):
    cash -= 5
    coins += 1

while (cash - 1 >= 0):
    cash -= 1
    coins += 1

print(coins)
